syft/lib.go

```diff
@@ -61,24 +64,54 @@ func IdentifyDistro(s scope.Scope) distro.Distro {
 	return d
 }
 
-// Catalog the given scope, which may represent a container image or filesystem. Returns the discovered set of packages.
-func CatalogFromScope(s scope.Scope) (*pkg.Catalog, error) {
+// Catalog the given source, which may represent a container image or filesystem. Returns the discovered set of packages.
+func CatalogFromScope(s source.Source) (*pkg.Catalog, error) {
 	log.Info("building the catalog")
 
 	// conditionally have two sets of catalogers
 	var catalogers []cataloger.Cataloger
-	switch s.Scheme {
-	case scope.ImageScheme:
+	switch s.Metadata.Scheme {
+	case source.ImageScheme:
 		catalogers = cataloger.ImageCatalogers()
-	case scope.DirectoryScheme:
+	case source.DirectoryScheme:
 		catalogers = cataloger.DirectoryCatalogers()
 	default:
-		return nil, fmt.Errorf("unable to determine cataloger set from scheme=%+v", s.Scheme)
+		return nil, fmt.Errorf("unable to determine cataloger set from scheme=%+v", s.Metadata.Scheme)
 	}
 
 	return cataloger.Catalog(s.Resolver, catalogers...)
 }
 
+// CatalogFromJSON takes an existing syft report and generates catalog primitives.
```

Comment:
Nit: The intended meaning of "catalog primitives" isn't obvious. This function returns a few different things, one of which is `*pkg.Catalog`, and other values as well. It might be worthwhile to spell out what's being returned, or, just state generically that this function deserializes a syft document into native syft objects.